CONVENTIONS.md

```diff
@@ -1,7 +1,38 @@
+In order to achieve our goal of wrapping [libc][libc] code in idiomatic rust
```

Comment:
Really minor: add a `# Conventions` or similar title up here, and make the lower headings h2 (`##`)